core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -611,24 +611,59 @@ public ClientFactoryBuilder http1MaxChunkSize(int http1MaxChunkSize) {
 
     /**
      * Sets the idle timeout of a socket connection. The connection is closed if there is no request in
-     * progress for this amount of time.
+     * progress for the given amount of time. By default, HTTP/2 PING frames do not prevent connection from
+     * closing. Use the method {@link ClientFactoryBuilder#idleTimeout(Duration, boolean)} to set whether to
+     * prevent connection from closing when an HTTP/2 PING frame is received.
      */
     public ClientFactoryBuilder idleTimeout(Duration idleTimeout) {
         requireNonNull(idleTimeout, "idleTimeout");
         checkArgument(!idleTimeout.isNegative(), "idleTimeout: %s (expected: >= 0)", idleTimeout);
         return idleTimeoutMillis(idleTimeout.toMillis());
     }
 
+    /**
+     * Sets the idle timeout of a socket connection and whether to reset idle timeout when an HTTP/2 PING
+     * frame is received.
+     */
+    public ClientFactoryBuilder idleTimeout(Duration idleTimeout, boolean keepAliveOnPing) {
```

Comment:
Let's add `@UnstableApi` to newly-added public methods in `ClientFactoryBuilder` and `ServerBuilder`.